### PR TITLE
Add doc explaining how to fix commits and commit messages

### PR DESCRIPTION
**Pull request by arpith**
Add docs explaining how to fix commits  …
1. Fixing commits and commit messages
2. Deleting commits
3. Squashing commits
4. Reordering commits

From https://help.github.com/articles/changing-a-commit-message/#rewriting-the-most-recent-commit-message and http://gitready.com/advanced/2009/03/20/reorder-commits-with-rebase.html

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@arpith, it looks like you've already signed the Dropbox CLA. Thanks!

File: docs/fixing-commits.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Fixing Commits
+This is mostly from https://help.github.com/articles/changing-a-commit-message/#rewriting-the-most-recent-commit-message
+## Fixing the last commit
+### Changing the last commit message
+1. `git commit --amend -m "New Message"`
+2. `git push -f`
+
+### Changing the last commit
+1. Make your changes to the files
+2. Add them with `git add .`
```
Comment (TomaszKolek) [inline docs/fixing-commits.md]: I would rather try to get used students to use `git add <file_name>`

File: docs/fixing-commits.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+## Fixing older commits
+### Changing commit messages
+1. `git rebase -i HEAD~5` (if, for example, you are editing the last five commits)
+2. For each commit that you want to change the message, change `pick` to `reword`
+3. Save, and change the commit messages
+4. `git push -f`
+
+### Deleting old commits
+1. `git rebase -i HEAD~n` where `n` is the number of commits you are looking at
+2. Delete the lines containing the commits you wish to erase, and save.
```
Comment (TomaszKolek) [inline docs/fixing-commits.md]: I'm not sure which version is better - remove the line or changing to `d` ?

File: docs/fixing-commits.md (hunk 1 above)
Comment (arpith) [inline docs/fixing-commits.md]: Fixed, thanks!

File: docs/fixing-commits.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+### Changing the last commit
+1. Make your changes to the files
+2. Add them with `git add .`
+3. `git commit --amend`
+4. `git push -f`
+
+## Fixing older commits
+### Changing commit messages
+1. `git rebase -i HEAD~5` (if, for example, you are editing the last five commits)
```
Comment (arpith) [inline docs/fixing-commits.md]: Changed :)

File: docs/fixing-commits.md (hunk 2 above)
Comment (arpith) [inline docs/fixing-commits.md]: Yeah drop/d sounds better - fixed! Thanks!

**Comment by arpith** [comment]
Thanks for your comments @TomaszKolek! I've made the changes :)

@timabbott could you take a look at this?

**Comment by krtkmj** [comment]
@arpith Can you please also add instructions for reordering git commit messages? Read more [here](http://gitready.com/advanced/2009/03/20/reorder-commits-with-rebase.html)

**Comment by arpith** [comment]
@krtkmj could you take a look now?

**Comment by krtkmj** [comment]
@arpith Looks good to me. Merging.

**Pull request merged**